### PR TITLE
consul-template 0.10.0 (new formula)

### PR DESCRIPTION
**Pull request by markcornick**

### CODE REVIEW

**Comment by markcornick** [comment]
Got Travis CI failures related to GitHub API rate limiting. The formula itself should be fine - I fixed one issue that brew audit found.

File: Library/Formula/consul-template.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    Language::Go.stage_deps resources, gopath/"src"
+
+    cd gopath/"src/github.com/hashicorp/consul-template" do
+      system "make"
+      bin.install "bin/consul-template"
+    end
+  end
+
+  test do
+    mktemp do
```
Comment (DomT4) [inline Library/Formula/consul-template.rb]: You're already in a temporary directory here. Don't need to create a new one.

File: Library/Formula/consul-template.rb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      File.open("test-template", "w") do |template|
```
Comment (DomT4) [inline Library/Formula/consul-template.rb]: Should be able to squash this into:

```ruby
(testpath/"test-template").write <<-EOS.undent
  {{"homebrew" | toTitle}}
EOS
```

File: Library/Formula/consul-template.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        template.puts '{{"homebrew" | toTitle}}'
+      end
+      exec "#{bin}/consul-template", "-once", \
```
Comment (DomT4) [inline Library/Formula/consul-template.rb]: Calling `system` rather than `exec` is generally preferred, unless the latter is necessary?

**Comment by markcornick** [comment]
All of @DomT4's comments should be addressed now.

File: Library/Formula/consul-template.rb [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    end
+  end
+
+  test do
+    (testpath/"test-template").write <<-EOS.undent
+      {{"homebrew" | toTitle}}
+    EOS
+    system "#{bin}/consul-template", "-once", \
+           "-template", "test-template:test-result"
+    assert_equal ["Homebrew\n"], File.open("test-result").readlines
```
Comment (bfontaine) [inline Library/Formula/consul-template.rb]: You can use `assert_equal "Homebrew\n", (testpath/"test-result").read`.

**Comment by bfontaine** [comment]
@markcornick Don’t worry about the Travis failure; the only build that matters here is the “default” one.

**Comment by markcornick** [comment]
Ok, done.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @markcornick; we appreciate it! :tada:

**Pull request closed without merging**